src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/util/ClientUtils.java

```diff
@@ -52,9 +52,9 @@ public static boolean hasValidLocalSignCert(ClientId clientId, List<CertificateI
             String ocspResponseStatus = null;
             try {
                 ocspResponseStatus = OcspUtils.getOcspResponseStatus(certificateInfo.getOcspBytes());
-            } catch (OcspUtils.OcspStatusExtractionException | RuntimeException e) {
+            } catch (OcspUtils.OcspStatusExtractionException e) {
                 log.error(ERROR_OCSP_EXTRACT_MSG + " for client: " + clientId.toString(), e);
-                return false;
+                break;
             }
             if (clientId.memberEquals(certificateInfo.getMemberId())
```

Comment:
Come to think of it, I would recommend moving this memberEquals check to be done as first thing in `hasValidLocalSignCert` for each certificateInfo. If memberEquals == false we don't need to inspect the cert any further, and can proceed to next candidate.

Now for each call to `hasValidLocalSignCert(client x)` it potentially scans through all certificates, and goes to `OcspUtils.getOcspResponseStatus()` to parse OCSP responses from bytes, even if the certificate has nothing to do with client x. For main clients list this can result in magnitude of `n*m` unnecessary getOcspResponseStatus() calls (n = number of clients, m = number of certs).

Reply:
Moved the comparison up in the method